spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -256,29 +396,63 @@ function sendFungibleTokens(
 
 ```typescript
 function onRecvPacket(packet: Packet) {
-  FungibleTokenPacketData data = packet.data
+  FungibleTokenPacketData data = unmarshal(packet.data)
   // construct default acknowledgement of success
   FungibleTokenPacketAcknowledgement ack = FungibleTokenPacketAcknowledgement{true, null}
   prefix = "{packet.sourcePort}/{packet.sourceChannel}/"
   // we are the source if the packets were prefixed by the sending chain
   source = data.denom.slice(0, len(prefix)) === prefix
+  // if we are forwarding the tokens along, we will move the coins to the forwardEscrowAddress to the source chain
+  // otherwise send directy to receiver defined by packet
+  if forwardingPath != "" {
+    // pass in denomination before receive to getNextHop
+    // i.e. denomination in packet
+    forwardPort, forwardChannel = getNextHop(forwardingPath, packet.denomination)
```

Comment:
Assuming we are unwinding a packet that went chain A -> chain B -> chain C and `forwardingPath` is `channels:origin` and denomination on chain C is `transfer/channel-3/transfer/channel-1/denom` (assuming `channel-3` is the destination channel on chain C and `channel-1` is the destination channel on chain B when the packet was sent from A to C), if the packet is sent from chain C and received on chain B, then `getNextHop` will return `transfer` and `channel-3`, right? But what we really want is `transfer` and `channel-1`, isn't it?